src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -76,75 +70,114 @@ using osp::Matrix3;
 using osp::Matrix4;
 using osp::Vector3;
 
-namespace
-{
 
 // Callback called for dynamic rigid bodies for applying force and torque
-static void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
+void SysNewton::cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
 {
     // Get context from Newton World
     auto const *const pWorldCtx = static_cast<ACtxNwtWorld*>(
             NewtonWorldGetUserData(NewtonBodyGetWorld(pBody)));
 
     // Get Associated entity
-    uintptr_t const userData = reinterpret_cast<uintptr_t>(
+    auto const userData = reinterpret_cast<uintptr_t>(
             NewtonBodyGetUserData(pBody));
-    ActiveEnt const ent = static_cast<ActiveEnt>(userData);
-
-    // Apply force
-
-    Vector3 force;
-    bool hasForce = false;
+    auto const bodyId = static_cast<BodyId>(userData);
```

Comment:
Are we using the pointer, casted to an int, as the Id?

That's fair i guess.

Are you certain that BodyId and uintptr_t will fit properly? Is that asserted somewhere?

Honestly might be worth considering making an adapter function to "hide" how someone converts from the return type of NewtonBodyGetUserData -> BodyId

Reply:
uintptr_t just has to be large enough. Anything in the code referred to as an Id denotes some instance starting at zero and is usually a uint32. I don't consider this important enough to need an assert.

I just realized `reinterpret_cast<void*>(bodyId)` used to set the user data elsewhere likely reads some extra bytes. I guess this highlights the need for such an adapter function.